integration_test/cases/browser/fullpage_screenshot_test.exs

```diff
@@ -0,0 +1,84 @@
+defmodule Wallaby.Integration.Browser.FullpageScreenshotTest do
+  use Wallaby.Integration.SessionCase, async: false
+
+  import Wallaby.SettingsTestHelpers
+
+  alias Wallaby.TestSupport.TestWorkspace
+
+  setup %{session: session} do
+    page =
+      session
+      |> visit("/")
+
+    {:ok, page: page}
+  end
+
+  test "taking fullpage screenshots", %{page: page} do
+    screenshots_path = TestWorkspace.generate_temporary_path()
+
+    ensure_setting_is_reset(:wallaby, :screenshot_dir)
+    Application.put_env(:wallaby, :screenshot_dir, screenshots_path)
+
+    [path] =
+      page
+      |> take_screenshot(name: "fullpage_test", full_page: true)
+      |> Map.get(:screenshots)
+
+    assert_in_directory(path, screenshots_path)
+    assert Path.basename(path) == "fullpage_test.png"
+    assert_file_exists(path)
+
+    # Verify the file is a valid PNG by checking the PNG signature
+    {:ok, file_content} = File.read(path)
```

Comment:
I would just use `File.read!` here, so if it fails, it fails with the right error and not a match error.